boa/contracts/vyper/vyper_contract.py

```diff
@@ -68,7 +68,98 @@
+
+    def verify(self, explorer: str) -> bool:
+        if explorer.lower() == "blockscout":
+            return self._verify_blockscout()
+        elif explorer.lower() == "etherscan":
+            return self._verify_etherscan()
+        else:
+            raise ValueError(f"Unsupported explorer: {explorer}")
+
+    def _verify_blockscout(self) -> bool:
+        api_key = os.getenv('BLOCKSCOUT_API_KEY')
+        if not api_key:
+            raise ValueError("BLOCKSCOUT_API_KEY not set in environment variables")
+
+        url = os.getenv('BLOCKSCOUT_API_URL', 'https://blockscout.com/poa/core/api')
+        
+        standard_json_input = {
+            "language": "Vyper",
+            "sources": {
+                "contract.vy": {
+                    "content": self.source_code
+                }
+            },
+            "settings": {
+                "optimizer": {
+                    "enabled": True
+                },
+                "outputSelection": {
+                    "*": ["evm.bytecode", "evm.deployedBytecode", "abi"]
+                }
+            }
+        }
+
+        params = {
+            "module": "contract",
+            "action": "verifysourcecode",
+            "addressHash": self.address,
+            "contractSourceCode": json.dumps(standard_json_input),
+            "name": "VerifiedContract",
+            "compilerVersion": self.compiler_version,
+            "optimization": "true",
+            "apikey": api_key
+        }
+        response = requests.post(url, params=params)
+        result = response.json()
+        return result.get("status") == "1"
+
+    def _verify_etherscan(self) -> bool:
+        api_key = os.getenv('ETHERSCAN_API_KEY')
+        if not api_key:
+            raise ValueError("ETHERSCAN_API_KEY not set in environment variables")
+
+        url = os.getenv('ETHERSCAN_API_URL', 'https://api.etherscan.io/api')
+        
+        standard_json_input = {
+            "language": "Vyper",
+            "sources": {
+                "contract.vy": {
+                    "content": self.source_code
+                }
+            },
+            "settings": {
+                "optimizer": {
+                    "enabled": True
+                },
+                "outputSelection": {
+                    "*": ["evm.bytecode", "evm.deployedBytecode", "abi"]
+                }
+            }
+        }
 
+        params = {
+            "module": "contract",
+            "action": "verifysourcecode",
+            "contractaddress": self.address,
+            "sourceCode": json.dumps(standard_json_input),
+            "codeformat": "solidity-standard-json-input",
+            "contractname": "contract.vy:VerifiedContract",
+            "compilerversion": self.compiler_version,
+            "optimizationUsed": "1",
+            "apikey": api_key
+        }
+        response = requests.post(url, params=params)
+        result = response.json()
+        return result.get("status") == "1"
```

Comment:
instead of returning a boolean, it would be better to just raise an error that includes the body returned by the explorer